cardinal/router/provider.go

```diff
@@ -0,0 +1,18 @@
+package router
+
+import (
+	"pkg.world.dev/world-engine/cardinal/types/message"
+	"pkg.world.dev/world-engine/sign"
+)
+
+//go:generate mockgen -source=provider.go -package mocks -destination=mocks/provider.go
+type Provider interface {
```

Comment:
Would appreciate a comment here on what this interface is used for

Reply:
oh, i understand the point of this now. would still love the comment, but yeah i think this is a great idea. we should probably do this for server and event hub as well.

Reply:
added